app/code/Magento/ProductAlert/Block/Email/AbstractEmail.php

```diff
@@ -212,18 +227,29 @@ public function getProductPriceHtml(
     }
 
     /**
-     * Retrieve product image
+     * Retrieve product image.
      *
-     * @param \Magento\Catalog\Model\Product $product
-     * @param string $imageId
+     * @param       $product
+     * @param       $imageId
      * @param array $attributes
+     *
      * @return \Magento\Catalog\Block\Product\Image
      */
     public function getImage($product, $imageId, $attributes = [])
     {
-        return $this->imageBuilder->setProduct($product)
+        $this->appEmulation->startEnvironmentEmulation(
+            $this->storeManager->getStore()->getId(),
+            \Magento\Framework\App\Area::AREA_FRONTEND,
+            true
+        );
+
+        $image = $this->imageBuilder->setProduct($product)
```

Comment:
Would be great to wrap this part with try-catch block, similar to this:
https://github.com/magento/magento2/blob/develop/app/code/Magento/Payment/Helper/Data.php#L218